test/integration_test.py

```diff
@@ -559,12 +559,13 @@ def _assert_catalog_complete(self,
 
     def _assert_catalog_empty(self, catalog: CatalogName):
         with self.subTest('catalog_empty', catalog=catalog):
-            hit_counts = {
-                entity_type: len(self._get_entities(catalog, entity_type))
+            entities = {
+                entity_type: self._get_entities(catalog, entity_type)
                 for entity_type in self.entity_types
             }
+            hit_counts = {entity_type: len(hits) for entity_type, hits in entities.items()}
             log.info('Hit counts are %r', hit_counts)
-            self.assertEqual(0, sum(hit_counts.values()))
+            self.assertEqual(0, sum(hit_counts.values()), ('Hits are %r', entities))
```

Comment:
- [ ] https://github.com/DataBiosphere/azul/pull/2364#discussion_r512324210

Reply:
Let's rethink this a bit. If the hit counts are all zero. The entities array has a predictable shape `{et: [] for et in self.entity_types}`. We can exploit this to get a listing of left-over entities for free, without the message tuple:

![image](https://user-images.githubusercontent.com/5143256/97660411-b5739200-1a2e-11eb-814c-53a937647f6c.png)